Mathlib/CategoryTheory/Limits/Shapes/FunctorToTypes.lean

```diff
@@ -0,0 +1,290 @@
+  app a x := by
+     cases x with
+     | inl x => exact τ₁.app a x
+     | inr x => exact τ₂.app a x
+  naturality _ _ _:= by
+    ext x
+    cases x with | _ => simp only [coprod, types_comp_apply, FunctorToTypes.naturality]
+
+@[simp]
+lemma coprod.desc_inl {F₁ F₂ : C ⥤ Type w} (τ₁ : F₁ ⟶ F) (τ₂ : F₂ ⟶ F) :
+    coprod.inl ≫ coprod.desc τ₁ τ₂ = τ₁ := rfl
+
+@[simp]
+lemma coprod.desc_inr {F₁ F₂ : C ⥤ Type w} (τ₁ : F₁ ⟶ F) (τ₂ : F₂ ⟶ F) :
+    coprod.inr ≫ coprod.desc τ₁ τ₂ = τ₂ := rfl
+
+variable (F G)
+
+/-- The binary cofan whose point is `coprod F G`. -/
+@[simps!]
+def binaryCoproductCocone : BinaryCofan F G :=
+  BinaryCofan.mk coprod.inl coprod.inr
+
+/-- `coprod F G` is a colimit cocone. -/
+@[simps]
+def binaryCoproductColimit : IsColimit (binaryCoproductCocone F G) where
+  desc (s : BinaryCofan F G) := coprod.desc s.inl s.inr
+  fac _ := fun ⟨j⟩ ↦ WalkingPair.casesOn j rfl rfl
+  uniq _ _ h := by
+    ext _ x
+    cases x with | _ => simp [← h ⟨WalkingPair.right⟩, ← h ⟨WalkingPair.left⟩]
+
+/-- `coprod F G` is a binary coproduct for `F` and `G`. -/
+def binaryCoproductColimitCocone : Limits.ColimitCocone (pair F G) :=
+  ⟨_, binaryCoproductColimit F G⟩
+
+/-- The categorical binary coproduct of type-valued functors is `coprod F G`. -/
+noncomputable def binaryCoproductIso : F ⨿ G ≅ coprod F G :=
+  colimit.isoColimitCocone (binaryCoproductColimitCocone F G)
+
+@[simp]
+lemma binaryCoproductIso_inl_comp_hom :
+    Limits.coprod.inl ≫ (binaryCoproductIso F G).hom = coprod.inl := by
+  simp only [binaryCoproductIso]
+  aesop
+
+@[simp]
+lemma binaryCoproductIso_inl_comp_hom_apply (a : C) (x : F.obj a) :
+    (binaryCoproductIso F G).hom.app a ((Limits.coprod.inl (X := F)).app a x) = .inl x :=
+  congr_fun (congr_app (binaryCoproductIso_inl_comp_hom F G) a) x
+
+@[simp]
+lemma binaryCoproductIso_inr_comp_hom :
+    Limits.coprod.inr ≫ (binaryCoproductIso F G).hom = coprod.inr := by
+  simp [binaryCoproductIso]
+  aesop
+
+@[simp]
+lemma binaryCoproductIso_inr_comp_hom_apply (a : C) (x : G.obj a) :
+    (binaryCoproductIso F G).hom.app a ((Limits.coprod.inr (X := F)).app a x) = .inr x :=
+  congr_fun (congr_app (binaryCoproductIso_inr_comp_hom F G) a) x
+
+@[simp]
+lemma binaryCoproductIso_inl_comp_inv :
+    coprod.inl ≫ (binaryCoproductIso F G).inv = (Limits.coprod.inl (X := F)) := rfl
+
+@[simp]
+lemma binaryCoproductIso_inl_comp_inv_apply (a : C) (x : F.obj a) :
+    (binaryCoproductIso F G).inv.app a (.inl x) = (Limits.coprod.inl (X := F)).app a x := rfl
+
+@[simp]
+lemma binaryCoproductIso_inr_comp_inv :
+    coprod.inr ≫ (binaryCoproductIso F G).inv = (Limits.coprod.inr (X := F)) := rfl
+
+@[simp]
+lemma binaryCoproductIso_inr_comp_inv_apply (a : C) (x : G.obj a) :
+    (binaryCoproductIso F G).inv.app a (.inr x) = (Limits.coprod.inr (X := F)).app a x := rfl
+
+variable {F G}
+
+/-- Construct an element of `(F ⨿ G).obj a` from an element of `F.obj a` -/
+noncomputable
+def coprodInl {a : C} (x : F.obj a) : (F ⨿ G).obj a :=
```

Comment:
```suggestion
abbrev coprodInl {a : C} (x : F.obj a) : (F ⨿ G).obj a :=
```
And similarly for `coprodInr`. Hopefully, this may ease automation in future uses of this API.